app/currency/tasks.py

```diff
@@ -1,26 +1,231 @@
 from celery import shared_task
 from django.core.mail import send_mail
+import requests
+from currency.utils import to_decimal
 
 
-# @shared_task
-# def print_hello_world():
-#     from time import sleep
-#     sleep(10)
-#     print('Hello world from celery process')
+def _get_privatbank_currencies():
+    url = 'https://api.privatbank.ua/p24api/pubinfo?json&exchange&coursid=5'
+    response = requests.get(url)
+    response.raise_for_status()
+    currencies = response.json()
+    return currencies
 
 
-# @shared_task
-# def print_hello_world(num):
-#     from time import sleep
-#     sleep(num)
-#     print(f'Hello world from celery process wait {num}')
+@shared_task()
+def parse_privatbank():
+    from currency.models import Rate
 
+    currencies = _get_privatbank_currencies()
 
-# @shared_task
-# def print_hello_world(rate_id):
-#     from currency.models import Rate
-#     rate = Rate.objects.get(id=rate_id)
-#     print(f'Got Rate with id: {rate.id}')
+    available_currencies_types = ('USD', 'EUR')
+
+    source = 'privatbank'
+
+    for curr in currencies:
+        currencies_type = curr['ccy']
+        if currencies_type in available_currencies_types:
+            buy = to_decimal(curr['buy'])
+            sale = to_decimal(curr['sale'])
+
+            previous_rate = Rate.objects.filter(source=source, type_curr=currencies_type).order_by('created').last()
+
+            if previous_rate is None or previous_rate.sale != sale or previous_rate.buy != buy:
+                # previous_rate is None or  # Check for no exist rate
+                # previous_rate.sale != sale or  # Check for changes related to last got data
+                # previous_rate.buy != buy
+                Rate.objects.create(
+                    type_curr=currencies_type,
+                    sale=sale,
+                    buy=buy,
+                    source=source,
+                )
+
+
+def _get_monobank_currencies():
+    url = 'https://api.monobank.ua/bank/currency'
+    response = requests.get(url)
+    response.raise_for_status()
+    currencies = response.json()
+    return currencies
+
+
+@shared_task()
+def parse_monobank():
+    from currency.models import Rate
+    currencies = _get_monobank_currencies()
+
+    available_currencies_raw = (840, 978)
+    second_type_curr = (980)
```

Comment:
`second_type_curr = (980)` -> `second_type_curr = 980`

Reply:
It is done in https://github.com/Story-TellerX/currency/pull/9/commits/cfdc84063186d0cf30cbfd8a093f9eaadaa447e4